File: app/src/main/java/com/salman/bitclock/data/database/SleepDao.java
package com.salman.bitclock.data.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.salman.bitclock.data.models.Sleep;

@Dao
public interface SleepDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Sleep sleep);

    @Query("SELECT * FROM sleep WHERE id = 1")
    LiveData<Sleep> getSleepSchedule();

}

File: app/src/main/kotlin/com/salman/bitclock/data/database/SleepDao.kt
package com.salman.bitclock.data.database

import androidx.room.*
import com.salman.bitclock.data.models.SleepSession
import kotlinx.coroutines.flow.Flow

@Dao
interface SleepDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(session: SleepSession): Long

    @Query("SELECT * FROM sleep_sessions ORDER BY start_time DESC")
    fun getAllSessions(): Flow<List<SleepSession>>

    @Query("SELECT * FROM sleep_sessions WHERE id = :sessionId")
    suspend fun getSessionById(sessionId: Long): SleepSession?

    @Delete
    suspend fun delete(session: SleepSession)
}
